app/app/src/main/res/navigation/mobile_navigation.xml

```diff
@@ -9,7 +9,12 @@
             android:id="@+id/navigation_playlist"
             android:name="me.bickositieff.raspio.ui.playlist.PlaylistFragment"
             android:label="@string/title_playlist"
-            tools:layout="@layout/fragment_playlist"/>
+            tools:layout="@layout/fragment_playlist">
+        <action
+                android:id="@+id/add_song"
+                app:destination="@id/navigation_song_select"
+                app:popUpTo="@id/navigation_playlist"/>
```

Comment:
popUpTo sets where to go when you navigate back from the selection, so the previous screen (playlist) is already the default.
```suggestion
                app:destination="@id/navigation_song_select"/>
```